app/src/main/java/org/wikipedia/dataclient/ServiceFactory.java

```diff
@@ -59,6 +65,29 @@ public static CoreRestService getCoreRest(@NonNull WikiSite wiki) {
         return s;
     }
 
+    public static EventService getAnalyticsRest(@NonNull StreamConfig streamConfig) {
+        DestinationEventService destinationEventService = streamConfig.getDestinationEventService();
+        if (destinationEventService == null) {
+            destinationEventService = DestinationEventService.ANALYTICS;
+        }
+
+        EventService s;
+        if (ANALYTICS_REST_SERVICE_CACHE.get(destinationEventService.getId()) != null) {
+            s = ANALYTICS_REST_SERVICE_CACHE.get(destinationEventService.getId());
+            if (s != null) {
+                return s;
+            }
+        }
+
+        String intakeBaseUriOverride = getEventPlatformIntakeUriOverride();
+
+        Retrofit r = createRetrofit(WikiSite.forLanguageCode(WikipediaApp.getInstance().getAppOrSystemLanguageCode()),
```

Comment:
Can we have a `WikiSite wiki` parameter and use that `wiki` to create a Retrofit instance?

e.g. `getAnalyticsRest(@NonNull WikiSite wiki, @NonNull StreamConfig streamConfig)`